dask_cloudprovider/aws/ecs.py

```diff
@@ -1434,7 +1439,7 @@ async def _cleanup_stale_resources():
                 role["Tags"] = (
                     await iam.list_role_tags(RoleName=role["RoleName"])
                 ).get("Tags")
-                if DEFAULT_TAGS.items() <= aws_to_dict(role["Tags"]).items():
+                if set(DEFAULT_TAGS.items()) <= set(aws_to_dict(role["Tags"]).items()):
```

Comment:
So this is the real change here, right? This seems to make sense to me.

Reply:
Yes, this fixes the core issue